examples/misc/combined_table/combined_table_common.py

```diff
@@ -0,0 +1,199 @@
+            live_filters: The filters to apply to the live table.
+        """
+        self._merge = merge
+        self._historical_raw = hist
+        self._live_raw = live
+        self._hist_filters = hist_filters
+        self._live_filters = live_filters
+        self._historical = None
+        self._live = None
+        self._combined = None
+
+    @property
+    def historical(self) -> Table:
+        """ The historical table."""
+        if not self._historical:
+            self._historical = self._historical_raw.where(self._hist_filters) \
+                if self._hist_filters else self._historical_raw
+
+        return self._historical
+
+    @property
+    def live(self) -> Table:
+        """ The live table."""
+        if not self._live:
+            self._live = self._live_raw.where(self._live_filters) \
+                if self._live_filters else self._live_raw
+
+        return self._live
+
+    @property
+    def combined(self) -> Table:
+        """ The combined table."""
+        if not self._combined:
+            self._combined = self._merge([self.historical, self.live])
+
+        return self._combined
+
+    @property
+    def apply(self) -> 'CombinedTable':
+        """ The CombinedTable with the filters fully applied."""
+        return CombinedTable(self._merge, self.historical, self.live)
+
+    @staticmethod
+    def _args_to_table_decorator(fn: Callable) -> Callable:
+        """ Decorator to convert function arguments from CombinedTable to Table objects."""
+
+        nargs = len(inspect.signature(fn).parameters)
+
+        if nargs == 0:
+            return fn
+
+        def wrapper(*args, **kwargs):
+            args_new = [arg.combined if isinstance(arg, CombinedTable) else arg for arg in args]
+            return fn(*args_new, **kwargs)
+
+        return wrapper
+
+    # If the method doesn't exist in this object, delegate it to the combined table.
+    # Methods are called with other CombinedTable objects as arguments, the arguments are converted to Table objects.
+    def __getattr__(*args):
+        obj = args[0]
+        fn = args[1]
+        c = obj.combined
+        f = getattr(c, fn)
+
+        # This is to handle @property methods and properties
+        if not callable(f):
+            return f
+
+        return CombinedTable._args_to_table_decorator(f)
+
+    @staticmethod
+    def _combine_filters(filters1: Sequence[str], filters2: Sequence[str]) -> Optional[Sequence[str]]:
+        """ Combine two sets of filters. If either filter is None, the other filter is returned."""
+
+        if isinstance(filters1, str):
+            filters1 = [filters1]
+
+        if isinstance(filters2, str):
+            filters2 = [filters2]
+
+        if not filters1 and not filters2:
+            return None
+        elif not filters1:
+            return filters2
+        elif not filters2:
+            return filters1
+        else:
+            return list(filters1) + list(filters2)
```

Comment:
Do we need to consider duplicates?

Reply:
I started to make this change but then thought it could be bad.  It is actually bad not making it as well as making it.

If duplicate filters are removed based on string inequality, consider these cases:
1. A UDF filter using a stateful function
2. `t.where(["X < random()", "X < random()"])`